kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/package-info.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+/**
+ *
+ * <h2 id='implementing'>Implementing Filters</h2>
+ *
+ * <h3 id='implementing.filterResults'>Filter Results</h3>
+ * <p>Filter implementation must return a {@link java.util.concurrent.CompletionStage} containing a
+ * {@link io.kroxylicious.proxy.filter.FilterResult} object. It is the job of FilterResult to convey what
+ * message is to forwarded to the next filter in the chain (or broker /client if at the chain's beginning
+ * or end).  It is also used to carry instructions such as indicating that the connection must be closed,
+ * or a message dropped.</p>
+ * <h3 id='implementing.deferringForward'>Deferring Forwards</h3>
+ * <p>If the filter returns a CompletionStage that is already completed normally, Kroxylicious will immediately
+ * perform the action described by the `FilterResult`.</p>
+ * <p>The filter may return a `CompletionStage` that is not yet completed. When this happens, Kroxylicious will pause
+ * reading from the downstream (the Client writes will eventually block), and it begins to queue up in-flight
+ * requests/responses arriving at the filter.  This is done so that message order is maintained.  Once the
```

Comment:
Should we call out at this level that this is an expensive operation in terms of performance and should be used sparingly?

Reply:
I've added the same paragraph that already appears in the Asciidoc.